### ISSUE TITLE
How to replicate specified arch images from DockerHub?

### ISSUE THREAD

**Issue by vndroid** [goharbor/harbor#19864]
The Docker Hub's Official Image has multi-arch sub-image. But I only need a part of them, how can I do automatic filtering while replicating?

Example: PHP has these arch sub-image:

```
linux /amd64
linux /arm/v6 
linux /arm/v7 
linux /arm64/v8 
linux /386 
linux /ppc64le 
linux /s390x 
```

I need only the `amd64` and `arm64/v8`, What can I do it?

**Comment by chlins** [comment]
Currently replication does not support filter single arch when the target image is a multi-arch image.

**Comment by jpylypiw** [comment]
Hi there 👋 
I really like the multi-arch feature, but in some cases a filter would be nice.
For example the mongodb image on DockerHub has images for linux and for windows. I am running a kubernetes cluster on linux nodes and only need the linux images. But the cluster is running on arm64 and amd64 nodes. So multi-arch is a good feature. The windows images now bloat my harbor instance and since I am running on slow-speed internet the replication is taking very long.
My idea is to provide a filter so in my case there would be two replication tasks. One for arm64 and one for amd64.

![image](https://github.com/goharbor/harbor/assets/12394156/cf826c48-4fc9-4763-948e-4fceef38c2ad)

**Comment by jidaojiuyou** [comment]
look at crane index filter (https://github.com/google/go-containerregistry/blob/main/cmd/crane/doc/crane_index_filter.md)

**Comment by kofj** [comment]
should we support arch filter?

**Comment by ludd98** [comment]
Yes, this is a needed feature. Some images have many arch so it takes up quite alot of space. I am only intrested in one arch for my clusters.

**Comment by vtmocanu** [comment]
I also would like such a filter as I have many replications and there are a lot of archs that I do not need and they take up a lot of space.
Thank you for considering this.

**Comment by cataglyphis** [comment]
+1

**Comment by Gauss23** [comment]
+1 from me!

**Comment by rnxmlu** [comment]
+2

**Comment by Viva-Linn** [comment]
+1

**Comment by stonezdj** [comment]
Harbor registry will validate the manifest's integrity at push, all sub-image and blob described in the manifest should exist in the registry before the manifest list(parent) push.
The problem is when partial of the manifest list's sub-images are replicated to harbor, in order to pass the integrity check, the final manifest content of the manifest list should be trimmed before push, like we did in the proxy cache. it will cause the replication job  always rerun when source/target artifact digest mismatch.

Suggest to use proxy cache instead of replication. because proxy cache always cache the arch it has been pulled.
